packages/collector/test/tracing/cloud/aws/dynamodb/app.js

```diff
@@ -0,0 +1,250 @@
+const delay = require('../../../../../../core/test/test_util/delay');
+
+const AWS = require('aws-sdk');
+const express = require('express');
+const logPrefix = `AWS DynamoDB (${process.pid}):\t`;
+AWS.config.update({ region: 'us-east-2' });
+const dynamoDB = new AWS.DynamoDB();
+const tableName = process.env.AWS_DYNAMODB_TABLE_NAME || 'nodejs-team';
+
+function log() {
+  /* eslint-disable no-console */
+  const args = Array.prototype.slice.call(arguments);
+  args[0] = `${logPrefix}${args[0]}`;
+  console.log.apply(console, args);
+  /* eslint-enable no-console */
+}
+
+var tableCreationParams = {
+  TableName: tableName,
+  KeySchema: [{ AttributeName: 'year', KeyType: 'HASH' }, { AttributeName: 'title', KeyType: 'RANGE' }],
+  AttributeDefinitions: [{ AttributeName: 'year', AttributeType: 'N' }, { AttributeName: 'title', AttributeType: 'S' }],
+  ProvisionedThroughput: {
+    ReadCapacityUnits: 5,
+    WriteCapacityUnits: 5
+  }
+};
+
+const itemKey = {
+  year: {
+    N: '2001'
+  },
+  title: {
+    S: 'new record'
+  }
+};
+
+const operationParams = {
+  createTable: tableCreationParams,
+
+  /**
+   * Deleting table is not part of the instrumentation scope, as seen on Java and Go implementations.
+   * However, we need to have this function in place to clean the table before starting tests
+   */
+  deleteTable: { TableName: tableName },
+  listTables: {},
+  scan: { TableName: tableName },
+  query: {
+    TableName: tableName,
+    KeyConditionExpression: '#yr = :yyyy',
+    ExpressionAttributeNames: {
+      '#yr': 'year'
+    },
+    ExpressionAttributeValues: {
+      ':yyyy': {
+        N: '2001'
+      }
+    }
+  },
+  getItem: {
+    TableName: tableName,
+    Key: itemKey
+  },
+  deleteItem: {
+    TableName: tableName,
+    Key: itemKey
+  },
+  putItem: {
+    TableName: tableName,
+    Item: itemKey
+  },
+  updateItem: {
+    TableName: tableName,
+    Key: itemKey,
+    AttributeUpdates: {
+      author: {
+        Value: {
+          S: 'Neil Gaiman'
+        }
+      }
+    }
+  }
+};
+
+const availableOperations = Object.keys(operationParams);
+
+const methods = {
+  CALLBACK: 'Callback',
+  PROMISE: 'Promise',
+  ASYNC: 'Async'
+};
+
+const availableMethods = Object.values(methods);
+
+const DynamoDBApi = {
+  runOperation(operation, method, withError) {
+    const originalOptions = operationParams[operation];
+    let options;
+    if (originalOptions) {
+      options = Object.assign({}, originalOptions);
+    }
+
+    if (withError) {
+      if (!options) {
+        options = {};
+      }
+      options.InvalidDynamoDBKey = '999';
+    }
+
+    return new Promise(async (resolve, reject) => {
+      let span;
+      let promise;
+      let promiseData;
+
+      switch (method) {
+        case methods.CALLBACK:
+          dynamoDB[operation](options, (err, data) => {
+            span = instana.currentSpan();
+            span.disableAutoEnd();
+
+            if (err) {
+              span.end(1);
```

Comment:
Related to the questionable disableAutoEnd mentioned above, I think all the manual span.end calls should not be necessary at all.